internal/controllers/topology/cluster/structuredmerge/dryrun.go

```diff
@@ -46,6 +48,22 @@ type dryRunSSAPatchInput struct {
 
 // dryRunSSAPatch uses server side apply dry run to determine if the operation is going to change the actual object.
 func dryRunSSAPatch(ctx context.Context, dryRunCtx *dryRunSSAPatchInput) (bool, bool, error) {
+	// Compute a request identifier.
+	// The identifier is unique for a specific request to ensure we don't have to re-run the request
+	// once we found out that it would not produce a diff.
+	// The identifier consists of: gvk, namespace and name of originalUnstructured and a hash of modifiedUnstructured.
+	// This ensures that we re-run the request as soon as either original or modified changes.
+	requestIdentifier, err := dryRunCtx.ssaRequestTracker.ComputeRequestIdentifier(dryRunCtx.originalUnstructured, dryRunCtx.modifiedUnstructured)
+	if err != nil {
+		return false, false, err
+	}
+
+	// Check if we already ran this request before by checking if the tracker already contains this identifier.
+	// Note: We only add an identifier to the tracker if the result of the dry run was no changes.
+	if exists := dryRunCtx.ssaRequestTracker.Has(requestIdentifier); exists {
+		return false, false, nil
+	}
+
```

Comment:
It would be great if we can expose this as a method of the tracker and thus keep request identity as internal of the ssa package e.g by providing something like `tracker.Get(original, modified)`

Reply:
I was thinking about this and we can do it, but it becomes tricky to use.

The problem is that we have to modify the `Add` func in the same way. But users have to be super careful that none of those two objects are modified between the Has and the Get call otherwise the cache key will be different.

(and usually the modified object is modified through the SSA call)

Reply:
Another option might be to pass in everything between the has and add call into the ssa func, roughly similar to once.Do()